File: Semester3/MAP/Seminare/Sem10/Sem10/Container/AbstractContainer.cs
namespace Sem10;

public abstract class AbstractContainer : Container
{
    protected List<Task> tasks = new List<Task>();
    
    public abstract Task remove();

    public void add(Task task)
    {
        tasks.Add(task);
    }

    public int size()
    {
        return tasks.Count;
    }

    public bool isEmpty()
    {
        return size() == 0;
    }
}

File: Semester3/MAP/laboratoare/lab2/TaskManager/src/Domain/AbstractContainer.java
package Domain;

import java.util.ArrayList;

public abstract class AbstractContainer implements Container{
    private ArrayList<Task> tasks = new ArrayList<>();

    public abstract Task remove();

    @Override
    public void add(Task task) {
        tasks.add(task);
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }

    @Override
    public int size() {
        return tasks.size();
    }

    protected ArrayList<Task> getTasks() {
        return tasks;
    }
}
